File: backend/app/schemas/product.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class ProductResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    merchant_id: UUID

    name: str
    sku: str

    description: str | None

    price: Decimal

    stock: int

    category: str


class ProductListResponse(BaseModel):
    products: list[ProductResponse]
    total: int

File: frontend/types/product.ts
export interface Product {
  id: string;
  merchant_id: string;

  name: string;
  sku: string;

  description: string | null;

  price: number;

  stock: number;

  category: string;
}

export interface ProductListResponse {
  products: Product[];
  total: number;
}